### PR TITLE
Olio/write_faults.py tests added

### PR DESCRIPTION
**Pull request by maxm17**
Olio/write_faults.py tests added

### CODE REVIEW

**Review by cflynn3 — APPROVED** [review]

File: tests/test_olio_write_faults.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
+import pandas as pd
+import pytest
+import resqpy.olio.write_faults as wf
+from pytest_mock import MockerFixture
+
+
+@pytest.mark.parametrize("test_df", [pd.DataFrame(), None])
+def test_no_data_in_faults(test_df):
+    # Act & Assert
+    with pytest.raises(AssertionError):
```
Comment (connortann) [inline tests/test_olio_write_faults.py:10]: I think a `ValueError` would be more appropriate than an `AssertionError` here: asserts are generally used to aid debugging internal logic, and shouldn't be raised by users.

**Review by connortann — CHANGES_REQUESTED** [review]
Suggest small change but otherwise LGTM

File: tests/test_olio_write_faults.py (hunk 1 above)
Comment (connortann) [reply tests/test_olio_write_faults.py:10]: Following a chat with @cflynn3  - let's park this change for a separate PR. If we change from asserts to Exceptions, we might as well do that consistently across the code.

**Review by connortann — APPROVED** [review]

File: tests/test_olio_write_faults.py (hunk 1 above)
Comment (maxm17) [reply tests/test_olio_write_faults.py:10]: I'm testing that the code produces an AssertionError. Do you want me to alter the write_faults.py file to produce a value error?

Comment (connortann) [reply tests/test_olio_write_faults.py:10]: That was my initial suggestion yes - but I think it's ok as it is for now, we could make that change at a later date.

Comment (maxm17) [reply tests/test_olio_write_faults.py:10]: I agree - I also want to add type hints and more detailed docstrings too

**Pull request merged**